lang/java/reef-runtime-azbatch/src/main/java/org/apache/reef/runtime/azbatch/client/AzureBatchDriverConfigurationProviderImpl.java

```diff
@@ -58,14 +65,23 @@ private AzureBatchDriverConfigurationProviderImpl(
       @Parameter(AzureBatchPoolId.class) final String azureBatchPoolId,
       @Parameter(AzureStorageAccountName.class) final String azureStorageAccountName,
       @Parameter(AzureStorageContainerName.class) final String azureStorageContainerName,
+      @Parameter(TcpPortList.class) final Set<Integer> tcpPortList,
+      final ContainerRegistryProvider containerRegistryProvider,
       final CommandBuilder commandBuilder) {
     this.jvmSlack = jvmSlack;
     this.azureBatchAccountUri = azureBatchAccountUri;
     this.azureBatchAccountName = azureBatchAccountName;
     this.azureBatchPoolId = azureBatchPoolId;
     this.azureStorageAccountName = azureStorageAccountName;
     this.azureStorageContainerName = azureStorageContainerName;
+    this.containerRegistryProvider = containerRegistryProvider;
+
     this.commandBuilder = commandBuilder;
+    this.tcpPortList = new HashSet<String>(tcpPortList.size());
```

Comment:
Why is this conversion from Integer to String necessary?

Reply:
ConfigurationModule.setMultiple only accepts a set of strings. That is why I had to convert from ints to strings.

---
In reply to: [199968189](https://github.com/sharathmalladi/reef/pull/6#discussion_r199968189) [](ancestors = 199968189)